Lib/sqlite3/__main__.py

```diff
@@ -0,0 +1,98 @@
+import sqlite3
+import sys
+
+from argparse import ArgumentParser
+from code import InteractiveConsole
+from textwrap import dedent
+
+
+def execute(c, sql):
+    try:
+        for row in c.execute(sql):
+            print(row)
+    except sqlite3.Error as e:
+        tp = type(e).__name__
+        try:
+            print(f"{tp} ({e.sqlite_errorname}): {e}")
+        except:
+            print(f"{tp}: {e}")
+
+
+class SqliteInteractiveConsole(InteractiveConsole):
+
+    def __init__(self, connection):
+        super().__init__()
+        self._con = connection
+        self._cur = connection.cursor()
+
+    def runsql(self, sql):
+        if not sqlite3.complete_statement(sql):
+            return True
+        execute(self._cur, sql)
+        return False
+
+    def runsource(self, source, filename="<input>", symbol="single"):
+        keywords = {
```

Comment:
Pattern matching would look better here but it's up to you.

Reply:
Yeah, but OTOH it's just a couple of lines. OTOH again, I need to take a look at pattern matching; I haven't had use for it so far, so I have not bothered looking at it (time is a scarce resource).